app/Actions/Photo/Strategies/AddDuplicateStrategy.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Actions\Photo\Strategies;
+
+use App\Exceptions\PhotoSkippedException;
+use App\Models\Logs;
+use App\Models\Photo;
+
+class AddDuplicateStrategy extends AddBaseStrategy
+{
+	public function __construct(AddStrategyParameters $parameters, Photo $existing)
+	{
+		parent::__construct($parameters, $existing);
+	}
+
+	public function do(): Photo
+	{
+		// At least update the existing photo with additional metadata if
+		// available
+		$this->hydrateMetadata();
+		if ($this->photo->isDirty()) {
```

Comment:
Yes, you are right. `save` checks for `isDirty` by itself. I just used it here so that the log message is only printed if something really needs to be updated. This resembles the old behaviour.